libs/iresearch/include/iresearch/analysis/wordnet_synonyms_tokenizer.cpp

```diff
@@ -0,0 +1,93 @@
+#include "wordnet_synonyms_tokenizer.hpp"
+
+#include <absl/strings/ascii.h>
+#include <absl/strings/str_replace.h>
+#include <absl/strings/str_split.h>
+
+#include <iresearch/analysis/pipeline_tokenizer.hpp>
+#include <iresearch/analysis/token_attributes.hpp>
+#include <string_view>
+#include <utility>
+
+namespace irs::analysis {
+
+sdb::ResultOr<WordnetSynonymsTokenizer::WordnetInput>
+WordnetSynonymsTokenizer::Parse(const std::string_view input) {
+  std::vector<std::string_view> lines = absl::StrSplit(input, '\n');
+
+  std::string_view last_syn_set_id = "";
+  size_t line_number{};
+
+  SynonymsGroups synset;
+  SynonymsMap mapping;
+
+  for (const auto& line : lines) {
+    line_number++;
+    if (line.empty())
+      continue;
+
+    const std::string_view syn_set_id = line.substr(2, 9);
+
+    // I couldn't find explicit guarantees that SynSet IDs must appear
+    // sequentially, but popular implementations rely on this assumption.
+    if (last_syn_set_id != syn_set_id) {
+      synset.push_back(syn_set_id);
+    }
+
+    const size_t start = line.find('\'') + 1;
+    const size_t end = line.rfind('\'');
```

Comment:
Given a WordNet line is a CSV, using on `,` for extracting values seems to be more reliable, e.g. what if we have a string with `'` inside it

Reply:
Updated logic